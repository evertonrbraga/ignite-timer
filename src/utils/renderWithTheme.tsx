import { ReactNode } from 'react'
import { BrowserRouter } from 'react-router-dom'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { defaultTheme } from 'styles/themes/default'

interface WrapperProps {
  children: ReactNode
}

function Wrapper({ children }: WrapperProps) {
  return (
    <BrowserRouter>
      <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
    </BrowserRouter>
  )
}

export function renderWithTheme(ui: ReactNode) {
  return render(ui, { wrapper: Wrapper })
}
