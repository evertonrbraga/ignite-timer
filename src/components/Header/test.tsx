import { screen, fireEvent } from '@testing-library/react'
import { renderWithTheme } from 'utils/renderWithTheme'
import { Header } from '.'

describe('<Header />', () => {
  it('should check if the Header component renders correctly', () => {
    const { container } = renderWithTheme(<Header />)
    expect(container.firstChild).toMatchSnapshot()
  })

  beforeEach(() => {
    renderWithTheme(<Header />)
  })

  it('should check if the Timer icon goes to the right path', () => {
    const timer = screen.getByTitle('Timer')
    fireEvent.click(timer)
    expect(window.location.pathname).toBe('/')
  })

  it('should check if the Scroll icon goes to the right path', () => {
    const scroll = screen.getByTitle('Hist√≥rico')
    fireEvent.click(scroll)
    expect(window.location.pathname).toBe('/history')
  })
})
