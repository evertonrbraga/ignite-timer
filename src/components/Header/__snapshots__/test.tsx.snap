// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should check if the Header component renders correctly 1`] = `
.c0 {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.c0 nav {
  display: flex;
  gap: 0.5rem;
}

.c0 nav a {
  width: 3rem;
  height: 3rem;
  display: flex;
  justify-content: center;
  align-items: center;
  border-top: 3px solid transparent;
  border-bottom: 3px solid transparent;
}

.c0 nav a:hover {
  border-bottom: 3px solid;
}

<header
  class="c0"
>
  <img
    alt=""
    src="test-file-stub"
  />
  <nav>
    <a
      aria-current="page"
      class="active"
      href="/"
      title="Timer"
    >
      <svg
        fill="currentColor"
        height="24"
        viewBox="0 0 256 256"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="256"
          width="256"
        />
        <circle
          cx="128"
          cy="128"
          fill="none"
          r="88"
          stroke="currentColor"
          stroke-miterlimit="10"
          stroke-width="16"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="128"
          x2="167.6"
          y1="128"
          y2="88.4"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="104"
          x2="152"
          y1="8"
          y2="8"
        />
      </svg>
    </a>
    <a
      class=""
      href="/history"
      title="HistÃ³rico"
    >
      <svg
        fill="currentColor"
        height="24"
        viewBox="0 0 256 256"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="256"
          width="256"
        />
        <path
          d="M200,176V64a23.9,23.9,0,0,0-24-24H40"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="104"
          x2="168"
          y1="104"
          y2="104"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="104"
          x2="168"
          y1="136"
          y2="136"
        />
        <path
          d="M22.1,80A24,24,0,1,1,64,64V192a24,24,0,1,0,41.9-16h112A24,24,0,0,1,200,216H88"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
        />
      </svg>
    </a>
  </nav>
</header>
`;
